test/runtests.jl

```diff
@@ -232,6 +240,41 @@ end
     end
 end
 
+function acceleratingFlow(N;T=Float64,perdir=(1,),jerk=4,gy=0,mem=Array)
+    NN = (N,N)
+    L = N
+    grav = 1
+    U = √(grav*L)
+    g(i,t) = i==1 ? t*jerk : gy
+    ν = 0.001
+    return WaterLily.Simulation(
+        NN, (U,0.), N; ν,g,U,Δt=0.001,perdir,T,mem
+    )
+end
+@testset "Flow.jl with time-varying body force" begin
+    for f∈arrays
+        N = 16
+        # test accelerating flow (using periodic condition)
+        jerk = 4; sim = acceleratingFlow(N;jerk,mem=f)
+        sim_step!(sim,1.0); timeExact = WaterLily.time(sim)
+        uFinal = sim.flow.U[1] + 0.5*jerk*timeExact^2
+        u = sim.flow.u |> Array
+        @test (
+            WaterLily.L₂(u[:,:,1].-uFinal) < 1e-4 &&
+            WaterLily.L₂(u[:,:,2].-0) < 1e-4
+        )
+        # test hydrostatic pressure field (using periodic condition)
+        N=8
+        jerk = 0; gy = 1; sim = acceleratingFlow(N;jerk,gy,mem=f)
+        sim_step!(sim,1.0); timeExact = WaterLily.time(sim)
+        p = sim.flow.p |> Array
+        BC!(p,perdir=sim.flow.perdir)
+        pe = copy(sim.flow.p) |> Array; apply!((x)->sim.flow.g(2,timeExact)*(x[2]-N/2),pe)
+        BC!(pe,perdir=sim.flow.perdir)
+        @test WaterLily.L₂(p .- pe) < 5e-3 # the error due to accumulation of pressure solver tolerance
+    end
+end
```

Comment:
I think this could be a lot smaller/cleaner. More like your example case. Then I think this will be ready to merge.

Reply:
Aha I just make a commit on it. I hope it is clean enough. I delete the hydrostatic test case. and make a lot of comments.